services/storage/src/simcore_service_storage/datcore_adapter/datcore_adapter_settings.py

```diff
@@ -0,0 +1,31 @@
+from typing import Optional
+
+from models_library.basic_types import PortInt, VersionTag
+from pydantic import AnyHttpUrl, Field, validator
+from settings_library.base import BaseCustomSettings
+
+
+class DatcoreAdapterSettings(BaseCustomSettings):
+    ENABLED: bool = True
+    HOST: str = "datcore-adapter"
+    PORT: PortInt = 8000
+    VTAG: VersionTag = Field(
+        "v0", description="Datcore-adapter service API's version tag"
+    )
+
+    endpoint: Optional[AnyHttpUrl] = None
+
+    @validator("endpoint", pre=True)
+    @classmethod
+    def auto_fill_endpoint(cls, v, values):
```

Comment:
for this type of "dependent" fields, a cached_property is a better approach. 

So, instead of creating an endpoint field + validator, do instead a single cached_property:

```python
class DatcoreAdapterSettings(BaseCustomSettings):
   # ... 
   @cached_property
   def endpoint(self) -> AnyHttpUrl:
         return AnyHttpUrl.build(
                scheme="http",
                host=self.DATCORE_ADAPTER_HOST,
                port=self.DATCORE_ADAPTER_PORT,
                path=self.DATCORE_ADAPTER_VTAG,
            )
```